refactor: fix TS errors & build `dist-typings`

DavideIadeluca opened the pull request:
<!--
IMPORTANT: We LOVE seeing new pull requests, they excite us every single time they are submitted! As we have an obligation to maintain a healthy code standard, quality, and extensions, we take sufficient time for reviews. Please do create a separate pull request per change/issue/feature; we will ask you to split bundled pull requests.
-->

**Fixes #0000**

**Changes proposed in this pull request:**
<!-- fill this out, mention the pages and/or components which have been impacted -->

**Reviewers should focus on:**
<!-- fill this out, ask for feedback on specific changes you are unsure about -->

**Screenshot**
<!-- include an image of the most relevant user-facing change, if any -->

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).

**Required changes:**

- [ ] Related [Flarum core extension PR's](https://github.com/flarum/core/pulls): (Omit this section if irrelevant)

DavideIadeluca committed:
chore: update lockfile

DavideIadeluca committed:
chore: edit package.json

DavideIadeluca committed:
chore; adjust tsconfig

DavideIadeluca committed:
chore: unused

DavideIadeluca committed:
chore: basic ts

DavideIadeluca committed:
chore: add shims

DavideIadeluca committed:
chore: basic ts

DavideIadeluca committed:
chore: unused

DavideIadeluca committed:
chore: return type

DavideIadeluca committed:
refactor: export

DavideIadeluca committed:
chore: wip ts refactor

DavideIadeluca committed:
chore: remove custom logic used for Google

dsevillamartin commented:
@DavideIadeluca If this were to be rebased, is it good to merge? Or whats the reasoning behind the draft status?

DavideIadeluca committed:
chore: yarn upgrade

StyleCIBot committed:
Apply fixes from StyleCI

DavideIadeluca committed:
Merge branch 'master' into di/full-ts

DavideIadeluca committed:
docs: backwards compat

DavideIadeluca committed:
chore: more TS

DavideIadeluca committed:
chore: type first & last login as date, serialized as string

DavideIadeluca committed:
chore: "unused" param

DavideIadeluca committed:
chore: rename and add interfaces

DavideIadeluca committed:
chore: return JSX.Element

dsevillamartin commented on js/src/forum/extenders/extendLoginSignup.tsx:
```diff
@@ -74,29 +86,33 @@ export default function () {
 
       // Refresh the list of providers
-      const newProviders = await this.store.find('linked-accounts');
+      const newProviders = this.store.all('linked-accounts') as LinkedAccount[];
```
Tried to test this, but LinkStatus doesn't open a popup for linking (as it should, as otherwise the redirect just is a blank page with a window.opener error)_ -- not an issue introduced by this PR. I think it was actually broken 2 years ago maybe? https://github.com/FriendsOfFlarum/oauth/commit/b8e75a5a6ce247acca6e0d8b962b97d7e526abe6#diff-bcce18dea0831749056fea8e3c6f476ba4c82dc19ea67564984c13066e764e79 🙈

DavideIadeluca replied:
Right, thanks for catching this. Fixed in https://github.com/FriendsOfFlarum/oauth/pull/84/commits/b689ee765572968665608247b581c7d8be22bf68. 

Reason for the change is `Property 'some' does not exist on type 'ApiResponseSingle<Model>'.`.

dsevillamartin commented on js/src/@types/shims.d.ts:
```diff
@@ -0,0 +1,17 @@
+
+declare module 'flarum/forum/ForumApplication' {
+  export default interface ForumApplication {
+    fof_oauth_linkingInProgress: boolean | undefined;
+    fof_oauth_linkingProvider: string | undefined;
+    fof_oauth_loginInProgress: boolean | undefined;
```
Just a nitpick, but I think I prefer

```suggestion
    fof_oauth_linkingInProgress?: boolean;
    fof_oauth_linkingProvider?: string;
    fof_oauth_loginInProgress?: boolean;
```

DavideIadeluca committed:
refactor: change typings

Co-authored-by: David Sevilla Martin <6401250+dsevillamartin@users.noreply.github.com>

DavideIadeluca committed:
chore: revert CSS changes

DavideIadeluca committed:
fix: refresh providers

dsevillamartin committed:
fix: use proper refresh of linked accounts

dsevillamartin committed:
fix: update google icon to match other logos

Google branding guidelines seems to allow icon-only now - don't remember seeing this previously. They also don't seem to care too much about size/padding, as long as it's visible and the text is not misleading.

https://developers.google.com/identity/branding-guidelines#custom-button
